crates/lib/src/config.rs

```diff
@@ -42,18 +43,28 @@ fn get_pantry_dir() -> io::Result<PathBuf> {
 }
 
 fn get_pkgx_dir() -> io::Result<PathBuf> {
-    if let Ok(env_dir) = env::var("PKGX_DIR") {
-        let path = PathBuf::from(env_dir);
+    if let Ok(path) = env::var("PKGX_DIR").map(PathBuf::from) {
         if !path.is_absolute() {
-            return Ok(env::current_dir()?.join(path));
+            Ok(env::current_dir()?.join(path))
         } else {
-            return Ok(path);
+            Ok(path)
+        }
+    } else {
+        let home = dirs_next::home_dir();
```

Comment:
I'd love to just move the unwrapping here, since, if this fails, life is pretty much broken. But the `map()` calls below aren't wrong.

Reply:
I figure if for some reason `HOME` is unset but `XDG_BLAH` are we shouldn't fail here. We will fail later though.